modules/calib3d/src/fisheye.cpp

```diff
@@ -127,15 +127,27 @@ void cv::fisheye::projectPoints(InputArray objectPoints, OutputArray imagePoints
     {
         Vec3d Xi = objectPoints.depth() == CV_32F ? (Vec3d)Xf[i] : Xd[i];
         Vec3d Y = aff*Xi;
-        if (fabs(Y[2]) < DBL_MIN)
-            Y[2] = 1;
-        Vec2d x(Y[0]/Y[2], Y[1]/Y[2]);
+
+        if(fabs(Y[2]) < 1e-8)
```

Comment:
@sergregory check if Y[2] close to zero (When theta_d is close to 90 degrees +-1e-8)